File: abdulah amin/agentra_user_frontend/lib/models/booking.dart
class Booking {
  final String id;
  final String packageId;
  final String packageTitle;
  final String userId;
  final int seats;
  final String travelDate;
  final double totalPrice;
  final String status;
  final String paymentMethod;
  final String? paymentStatus;
  final String? packageImage;
  final DateTime createdAt;

  Booking({
    required this.id,
    required this.packageId,
    required this.packageTitle,
    required this.userId,
    required this.seats,
    required this.travelDate,
    required this.totalPrice,
    required this.status,
    required this.paymentMethod,
    this.paymentStatus,
    this.packageImage,
    required this.createdAt,
  });

  factory Booking.fromJson(Map<String, dynamic> json) {
    return Booking(
      id: json['_id'] ?? json['id'] ?? '',
      packageId: json['packageId'] is Map ? json['packageId']['_id'] : (json['packageId'] ?? ''),
      packageTitle: json['packageId'] is Map ? json['packageId']['title'] : (json['packageTitle'] ?? 'Unknown Package'),
      userId: json['userId'] ?? '',
      seats: json['seats'] ?? 1,
      travelDate: json['travelDate'] ?? '',
      totalPrice: (json['totalAmount'] ?? json['totalPrice'] ?? 0).toDouble(),
      status: json['status'] ?? 'pending',
      paymentMethod: json['paymentMethod'] ?? 'CARD',
      paymentStatus: json['paymentStatus'],
      packageImage: json['packageId'] is Map ? json['packageId']['image'] : json['packageImage'],
      createdAt: json['createdAt'] != null 
          ? DateTime.parse(json['createdAt'])
          : DateTime.now(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'packageId': packageId,
      'packageTitle': packageTitle,
      'userId': userId,
      'seats': seats,
      'travelDate': travelDate,
      'totalPrice': totalPrice,
      'status': status,
      'paymentMethod': paymentMethod,
      'paymentStatus': paymentStatus,
      'createdAt': createdAt.toIso8601String(),
    };
  }

  String getStatusText() {
    switch (status.toLowerCase()) {
      case 'confirmed':
        return 'Confirmed';
      case 'pending':
        return 'Pending';
      case 'cancelled':
        return 'Cancelled';
      case 'completed':
        return 'Completed';
      default:
        return 'Unknown';
    }
  }
}

File: Agentra/abdulah amin/agentra-backend/src/models/Booking.js
const mongoose = require('mongoose');

const bookingSchema = new mongoose.Schema({

  // -------- Relationships --------
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },

  agentId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Agent',
    required: true
  },

  packageId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Package',
    required: true
  },

  // -------- Travel Details --------
  seats: {
    type: Number,
    required: true
  },

  travelDate: {
    type: Date,
    required: true
  },

  // -------- Payment --------
  totalAmount: {
    type: Number,
    required: true
  },

  paymentStatus: {
    type: String,
    enum: ['PENDING', 'PAID', 'REFUNDED'],
    default: 'PENDING'
  },

  paymentMethod: {
    type: String,
    enum: ['JAZZCASH', 'BANK'],
    default: 'JAZZCASH'
  },

  // -------- Booking State --------
  status: {
    type: String,
    enum: ['CONFIRMED', 'CANCELLED', 'COMPLETED'],
    default: 'CONFIRMED'
  },

  // -------- System Tracking --------
  cancellationReason: String,
  refundStatus: {
    type: String,
    enum: ['NONE', 'REQUESTED', 'APPROVED', 'REJECTED'],
    default: 'NONE'
  }

}, { timestamps: true });

module.exports = mongoose.model('Booking', bookingSchema);
